recipes/yarn.rb

```diff
@@ -23,10 +23,40 @@
   dest "#{home}/#{node['flink']['user']}/flink.jar"
 end
 
-hops_hdfs_directory "#{node['flink']['home']}/conf/flink-conf.yaml" do
-  action :put_as_superuser
-  owner node['flink']['user']
-  group node['hops']['group']
-  mode "1775"
-  dest "#{home}/#{node['flink']['user']}/flink-conf.yaml"
+
+beamjobserver_jar=File.basename(node['flink']['beamjobserver_jar']['url'])
+  remote_file "#{Chef::Config['file_cache_path']}/#{beamjobserver_jar}" do
+    source node['flink']['beamjobserver_jar']['url']
+    owner node['flink']['user']
+    group node['hops']['group']
+    mode "1775"
+    action :create
+  end
+
+  hops_hdfs_directory "#{Chef::Config['file_cache_path']}/#{beamjobserver_jar}" do
+    action :replace_as_superuser
+    owner node['flink']['user']
+    group node['hops']['group']
+    mode "1755"
+    dest "/user/#{node['flink']['user']}/#{beamjobserver_jar}"
+  end
+
+
+ruby_block "hadoop_glob" do
+  block do
+    Chef::Resource::RubyBlock.send(:include, Chef::Mixin::ShellOut)
+    node.default['hadoop_glob'] = shell_out("#{node['hops']['bin_dir']}/hadoop classpath --glob  | tr -d '\n'").stdout
+  end
+  action :create
 end
+
+template "#{node['flink']['home']}/conf/sdk_worker.sh" do
```

Comment:
Is this recipe supposed to be run on every machine (like `spark::yarn`)? if so, could you please wrap all the `hops_hdfs_directory` in an if?

Reply:
No, flink is installed on one machine at the moment. Flink internally ships the `flink.jar` as a local resource. Now we added support in Flink for custom local resources that are stored in hdfs, so I use that for the `sdk_worker.sh` script.